samples/nrf5340/netboot/src/main.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr/types.h>
+#include <sys/printk.h>
+#include <pm_config.h>
+#include <fw_info.h>
+#include <fprotect.h>
+#include <bl_storage.h>
+#include <bl_boot.h>
+#include <bl_validation.h>
+#include <dfu/pcd.h>
+#include <device.h>
+
+#define FLASH_NAME DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL
+
+void main(void)
+{
+	int err = fprotect_area(PM_B0N_ADDRESS, PM_B0N_SIZE);
+	struct device *fdev = device_get_binding(FLASH_NAME);
+
+	if (err) {
+		printk("Failed to protect b0n flash, cancel startup\n\r");
+		goto failure;
+	}
+
+	uint32_t s0_addr = s0_address_read();
+	bool valid = false;
+
+	if (pcd_fw_copy_status_get() == PCD_STATUS_COPY) {
+		err = pcd_fw_copy(fdev);
```

Comment:
We could run `bl_validate_firmware(s0_addr, cmd->addr);` here to make sure we won't be bricking the network core.

Reply:
MCUBoot has already done a full check, so I don't think that would give any value. We know that data is as expected at the location pointed to by the cmd.

Reply:
MCUboot hasn't done this exact check. It's a bit neurotic, but not without value. Better safe than sorry kind of thing.

Reply:
Check now, I might have brokne the functionality, but will fix that tomorrow.